index.js

```diff
@@ -1,14 +1,21 @@
 const compiler = require('vue-template-compiler');
 const transpile = require('vue-template-es2015-compiler');
+const parse5 = require('parse5');
+const consolidate = require('consolidate');
+const spawnSync = require('child_process').spawnSync;
+const pathJoin = require('path').join;
 
 const exportsTarget = '((module.exports.default || module.exports).options || module.exports.default || module.exports)';
 
 module.exports = function ({ content, filename, sourceMap }) {
-  const startOfScript = content.indexOf('<script>') + 8;
-  const endOfScript = content.lastIndexOf('</script>');
-  if (startOfScript < 8 || endOfScript < 0 || endOfScript < startOfScript){
+  const documentFragment = parse5.parseFragment(content, {locationInfo: true});
```

Comment:
I haven't used parse5 before but it looks like a nice fit for this. My only concern is that it will parse the entire template, so if you are loading a particularly large component it could really slow things down, whereas a simple regex or something would probably be quicker.